### PR TITLE
Added unnatural instructions dataset

### PR DESCRIPTION
**Pull request by eitan3**
Added unnatural instructions dataset from 
https://github.com/orhonovich/unnatural-instructions 
https://www.tensorflow.org/datasets/catalog/unnatural_instructions

Include both instances data & reformulations data and cleaned the data little bit so finetune.py won't crash

### CODE REVIEW

**Comment by tloen** [comment]
Nice, have you tried training on this?

**Comment by eitan3** [comment]
Im training on it right now, combined this dataset and the cleaned dataset. 
66% first epoch (18 hours on 3090 for one epoch), train on 13B model.

**Comment by wassname** [comment]
Kind of unrelated, but mildly interesting.  Another good dataset is [SHP](https://huggingface.co/datasets/stanfordnlp/SHP). It's good because it's natural dialogue from good Subreddits and it's large.

I believe LLaMa wasn't trained on reddit, so the supervision might also fill a blindspot.

**Comment by eitan3** [comment]
I don't know what to tell you, for me its give better result, but I trained with the clean dataset too. 
Right now I'm working on adding chat dataset too, and for now I'm training only on the chat dataset.

**Comment by wassname** [comment]
Interesting! So to be clear, you trained on both alpaca_cleaned and unnatural instructions, and the results are qualitatively better than just fine-tuning with alpaca_cleaned. Am I understanding you correctly?

What chat dataset are you adding?

**Comment by eitan3** [comment]
Yeah, you're right. I got better answers, and the loss was much lower (I didn't save it.. next time I'll use weights and biases).
Now i'm trying to use helpful-base from https://huggingface.co/datasets/Anthropic/hh-rlhf
I manage to make it as chatbot but now its refuse to give me some basic answers. so next step is to add more data from anthropic's dataset and combine with alpaca_cleaned and unnatural instructions datasets. I just need to find a good way to convert the instructions to chat messages.

I also changed the template of the prompt (changed the output of generate_prompt function), and I think it give worse results..
yesterday I trained on alpaca_cleaned and unnatural instructions with different prompt and its was much worse than alpaca_cleaned and unnatural instructions and the original prompt

**Comment by wassname** [comment]
>  refuse to give me some basic answers

It's quite frustrating when it does this. One of the advantages of training your own model is to avoid "As a large language model blah blah" but it seems to somehow remain in some of the Lora model's I've tried. I'm hoping to try with SHP as well as HH-RLHF. And once `oasst` is released (from open-assistant) that too.

**Comment by eitan3** [comment]
I'm looking forward to open-assistant as well, hope it will release soon.
I think the magic of avoiding this kind of messages is in the prompt template (generate_prompt function) and the instruction you give him. The prompt I tried for the chatbot is
`
    out = f""""""
    out += f"""You are the bot in the conversation. Continue the conversation and answer shortly!"""
    out += f"""----"""
    out += f"""\n\n"""
    out += f"""{chat_messages}"""
    out += f"""\n\n"""
    out += f"""----"""
    out += f"""\n\n"""
    out += f"""### Response: """
    if output:
        out += f"""{output}"""
`

**Comment by doxav** [comment]
A more recent approach from Anthropic: https://github.com/anthropics/ConstitutionalHarmlessnessPaper seems a better approach

**Comment by eitan3** [comment]
Closed cause found too much duplications, need to filter the dataset..
I'll check the new approach by Anthropic. right now Im playing with different datasets
